airbyte-integrations/connectors/source-kafka/src/main/java/io/airbyte/integrations/source/kafka/mediator/KafkaMediatorImpl.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.source.kafka.mediator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.integrations.source.kafka.KafkaConsumerRebalanceListener;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KafkaMediatorImpl<V> implements KafkaMediator<V> {
```

Comment:
See other comment about "impl" naming.

Reply:
In this case the interface serves a purpose... that might be a design smell. If you look at the generator tests, we are providing test implementations of the meditator - instead of using mocks. In general, I prefer that approach, but it forces you to introduce an interface that might not be needed.

In this case - just for the shake of convenience - I think we should rename KafkaMediatorImpl to DefaultKafkaMediator.

Reply:
One thing I don't like - but I will leave for a future PR - is the fact that the mediator is finding and subscribing to the topics in the constructor. I think we should use a builder pattern here.

Reply:
> In this case the interface serves a purpose... that might be a design smell. If you look at the generator tests, we are providing test implementations of the meditator - instead of using mocks. In general, I prefer that approach, but it forces you to introduce an interface that might not be needed.
> 
> In this case - just for the shake of convenience - I think we should rename KafkaMediatorImpl to DefaultKafkaMediator.

OK, I understand; for my experience in these cases there's the chance that the test is "telling me" that there's some issues with the design of the objects that I am trying to test. Therefore, this interface issue, is just a sympthom.
But that's fine, this could be an analysis to postpone for a future PR 👍🏻

Reply:
> One thing I don't like - but I will leave for a future PR - is the fact that the mediator is finding and subscribing to the topics in the constructor. I think we should use a builder pattern here.

I agree 👍🏻